slack_rules/slack_app_access_expanded.py

```diff
@@ -0,0 +1,46 @@
+from panther_base_helpers import deep_get, slack_alert_context
+
+ACCESS_EXPANDED_ACTIONS = [
+    "app_scopes_expanded",
+    "app_resources_added",
+    "app_resources_granted",
+    "bot_token_upgraded",
+]
+
+
+def rule(event):
+    return event.get("action") in ACCESS_EXPANDED_ACTIONS
+
+
+def title(event):
+    return f"Slack App [{deep_get(event, 'entity', 'app', 'name')}] " \
+           f"Access Expanded by [{deep_get(event, 'actor', 'user', 'name')}]"
+
+
+def alert_context(event):
+    context = slack_alert_context(event)
+
+    # Diff previous and new scopes
+    new_scopes = deep_get(event, "details", "new_scopes", default=[])
+    prv_scopes = deep_get(event, "details", "previous_scopes", default=[])
+
+    context["scopes_added"] = [x for x in new_scopes if x not in prv_scopes]
```

Comment:
very nice. future self thanks you for the alpha grouping of added and removed items 👍